### ISSUE TITLE
GPU Implementation?

### ISSUE THREAD

**Issue by dhariyalbhaskar** [ljvmiranda921/pyswarms#47]
How can I use this package with Keras in optimiser class?
Any thoughts on GPU implementation?

**Comment by ljvmiranda921** [comment]
Hi, I've realized you're the same person in Issue #19 

As of now, implementing this package on GPU (perhaps via CUDA) is not yet being considered in the Development Roadmap for v.0.2.0 (#5). In all honesty I don't have any experience in CUDA 😞 . If you can help and show us a lead, then perhaps we can add this feature for v.0.3.0.

For your first question, what you can do is take the `update_*` and `optimize` methods of a PSO optimizer from pyswarms, and use these methods to populate a custom-class that inherits from Keras' `Optimizer` base class. 👍

**Comment by dhariyalbhaskar** [comment]
@ljvmiranda921 I have written a GPU PSO code with pyCUDA. I will share with you once you start v.0.3.0.
Thanks.

**Comment by ljvmiranda921** [comment]
Awesome! Okay, I will ping you once we've started tackling this feature. Thanks a lot!

**Comment by ljvmiranda921** [comment]
Hmmm, I just realized that maybe I can leverage Pytorch (as a substitute to numpy) to take advantage of GPU computations. This is just an idea in the clouds but I'll cook something up this winter vacation.
